scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+                                                                                                                          BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll on key '" + currentKey + "' result is: " + answer,
+                                   Level.DEBUG);
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
+            throw new ClosureEvaluationException("Exception when evaluating closure : " +
+                                                 StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    @Override
+    public <T extends Serializable> void conditionalReplaceAll(String originator, TaskId taskid, String channel,
+            String predicate, String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        String currentKey = null;
+        try {
+            Channel<T> chosenChannel = getChannel(channel);
+            for (Map.Entry<String, T> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                T value = chosenChannel.get(currentKey);
+                if (evaluateClosure(predicate, BiPredicate.class).test(currentKey, value)) {
+
+                    T answer = chosenChannel.compute(currentKey,
+                                                     (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(thenRemappingFunction,
+                                                                                                                          BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Then) on key '" + currentKey + "' result is: " + answer,
+                                   Level.DEBUG);
+                } else {
+                    T answer = chosenChannel.compute(currentKey,
+                                                     (BiFunction<? super String, ? super T, ? extends T>) evaluateClosure(elseRemappingFunction,
+                                                                                                                          BiFunction.class));
+                    logWithContext(taskid,
+                                   channel,
+                                   "Run conditionalReplaceAll(Else) on key '" + currentKey + "' result is: " + answer,
+                                   Level.DEBUG);
+
+                }
+            }
+            commitIfNeeded(channel);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)